clients/client-python/gravitino/client/gravitino_client_base.py

```diff
@@ -57,16 +65,49 @@ def load_metalake(self, ident: NameIdentifier) -> GravitinoMetalake:
 
         return GravitinoMetalake(metalake_response.metalake(), self._rest_client)
 
-    def get_version(self) -> GravitinoVersion:
+    def check_version(self):
+        """Check the compatibility of the client with the target server.
+
+        Raises:
+            GravitinoRuntimeException If the client version is greater than the server version.
+        """
+        server_version = self.get_server_version()
+        client_version = self.get_client_version()
+
+        if client_version > server_version:
+            raise GravitinoRuntimeException(
+                "Gravitino does not support the case that "
+                "the client-side version is higher than the server-side version."
+                f"The client version is {client_version.version()}, and the server version {server_version.version()}"
+            )
+
+    def get_client_version(self) -> GravitinoVersion:
+        """Retrieves the version of the Gravitino Python Client.
+
+        Returns:
+            A GravitinoVersion instance representing the version of the Gravitino Python Client.
+        """
+        version = pkg_resources.require("gravitino")[0].version
+        git_commit = (
+            subprocess.check_output(["git", "rev-parse", "HEAD"])
+            .decode("ascii")
+            .strip()
+        )
+        compile_date = datetime.now().strftime("%d/%m/%Y %H:%M:%S")
+
+        return GravitinoVersion(VersionDTO(version, compile_date, git_commit))
```

Comment:
Does this mean that we will get `git_commit` and `compile_data` from git command, does it assume that we should have a git repo beforehand?

I think it is not aligned with Java's implementation, In Java gradle will write a file and pack into the jar when compiling, the version info is read from the file when using, so that we don't rely on a git repo. Here in Python with this assumption it is hard to use without git repo, am I right?

Reply:
Yes, I forgot to consider this part, I think I will create the file in `build` task and read from it.

Reply:
Yeah, that would be better.